### PR TITLE
InvalidCastException in _browser.Close()

### PR DESCRIPTION
**Pull request by usarskyy**
This change is a fix for common issue when a test fails because of the InvalidCastException inside _browser.Close() method.
Please, include this fix in your next release

### CODE REVIEW

File: SourceCode/FluentAutomation.WatiN/AutomationProvider.cs [hunk 1]
```diff
@@ -30,8 +32,21 @@ public override void Authenticate(string username, string password)
...
+			if (_jsAlerDialogHandler != null)
+			{
+				_browser.RemoveDialogHandler(_jsAlerDialogHandler);
+				_browser.DialogWatcher.Clear();
+				_jsAlerDialogHandler = null;
+			}
+
+			if(_browser is Automation.Core.IE)
+			{
+				((SHDocVw.WebBrowser)((Automation.Core.IE) _browser).InternetExplorer).Stop();
```
Comment (usarskyy) [inline SourceCode/FluentAutomation.WatiN/AutomationProvider.cs:44]: Stops content loading. It is optional but to be on the safe side I do it before closing the browser

File: SourceCode/FluentAutomation.WatiN/AutomationProvider.cs [hunk 2]
```diff
@@ -30,8 +32,21 @@ public override void Authenticate(string username, string password)
...
+				((SHDocVw.WebBrowser)((Automation.Core.IE)_browser).InternetExplorer).Quit();
```
Comment (usarskyy) [inline SourceCode/FluentAutomation.WatiN/AutomationProvider.cs:45]: Helps to solve problem with "InvalidCastException" when a dialog is opened or browser still loads some data.

File: SourceCode/FluentAutomation.WatiN/AutomationProvider.cs [hunk 3]
```diff
@@ -30,8 +32,21 @@ public override void Authenticate(string username, string password)
 
         public override void Cleanup()
         {
-            _browser.Close();
-            _browser = null;
+			if (_jsAlerDialogHandler != null)
```
Comment (usarskyy) [inline SourceCode/FluentAutomation.WatiN/AutomationProvider.cs:35]: This if-statement is optional, but to be sure that we don't leave any references I try to cleanup everything

**Comment by stirno** [comment]
Basic idea committed after CI tests finally running. Sorry for the delay on this, crazy few months.

**Pull request closed without merging**